contracts/services/rebalancer/src/rebalance.rs

```diff
@@ -119,6 +122,23 @@ pub fn execute_system_rebalance(
 
         last_addr = Some(account.clone());
 
+        // Before rebalancing, verify the account is using a whitelisted code id
+        let account_code_id = deps
+            .querier
+            .query_wasm_contract_info(account.clone())?
+            .code_id;
+
+        if !whitelist.contains(&account_code_id) {
+            // Save to the paused config
+            PAUSED_CONFIGS.save(
+                deps.storage,
+                account.clone(),
+                &PauseData::new_not_whitelisted_account_code_id(env, account_code_id, &config),
+            )?;
+            // remove from active configs
+            CONFIGS.remove(deps.storage, account.clone());
+        }
```

Comment:
is there a reason not to `continue;` here if we confirm that account is non-whitelisted?

Reply:
Good catch, fixed.